laboratory-servlet-api-integration/src/test/java/com/wixpress/petri/laboratory/ResolversTest.scala

```diff
@@ -0,0 +1,89 @@
+package com.wixpress.petri.laboratory
+
+import javax.servlet.http.{Cookie, HttpServletRequest}
+
+import org.specs2.mutable.SpecificationWithJUnit
+import org.specs2.specification.Scope
+import org.springframework.mock.web.MockHttpServletRequest
+
+class ResolversTest extends SpecificationWithJUnit {
+
+  trait Context extends Scope {
+    val aDefaultResolution = "default"
+    val aFilterParam = FilterParameters.Language
+    val someValue = "someVal"
+
+    val resolver = new Resolver {
+      override def defaultResolution(request: HttpServletRequest): String = aDefaultResolution
+      override val filterParam: FilterParameters.Value = aFilterParam
+    }
+
+    val request = new MockHttpServletRequest
+  }
+  "resolve by default resolution when configs empty" in new Context {
+    val config = FilterParametersExtractorsConfig()
+    resolver.resolve(request, config) must beEqualTo (aDefaultResolution)
+  }
+
+  "resolve by header 'SOME_HEADER' when configured" in new Context {
+    val config = FilterParametersExtractorsConfig(Map(aFilterParam.toString -> List(("Header", "SOME_HEADER"))))
+
+    request.addHeader("SOME_HEADER", someValue)
+
+    resolver.resolve(request, config) must beEqualTo (someValue)
+  }
+
+  "resolve by 'SOME_COOKIE_NAME' cookie if filterParamConfig defines filterParam Country cookie" in new Context {
+    val config = new FilterParametersExtractorsConfig(
+      Map(aFilterParam.toString -> List(("Cookie", "SOME_COOKIE_NAME"))))
+
+    request.setCookies(new Cookie("SOME_COOKIE_NAME", someValue))
+
+    resolver.resolve(request, config) must beEqualTo(someValue)
+  }
+
+  "resolve by param 'SOME_PARAM' when configuration before that are not present" in new Context {
```

Comment:
maybe this test should test only the param configuration?

the 'resolve by order' part seems to be covered by the test just below